[FIX] website: fix "over the content" option not saved

bvr-odoo opened the pull request:
Since this commit [1], when you select the Header position option and
set it to "Over The Content", the change is not saved if you did not
edit something else in the page too.

Indeed, now that the "save" works correctly (and that it no longer saves
the page every time). It does not detect the change made by this option
which is to add a class on the '#wrapwrap' element. (The editor looks
for changes inside the wrapwrap but not on it).

With this commit, we add the 'o_dirty' class on the '#wrap' when a page
option is changed in order to notify the editor that the page should be
saved when the "save" button is clicked.

[1]: https://github.com/odoo/odoo/commit/bab673488e185ddd7792aedecc3870663290fed3

task-2871426

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:




--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

bso-odoo reviewed (approved):
LGTM
I guess it would be overkill to try put the `o_dirty` marker specifically on the header or the footer - and it would not really be correct given that the changes are only "about them" but not actually "on them".
Not sure if you want to add a "Steps to reproduce" in the commit message - given it is already very well described.

bvr-odoo commented:
> LGTM I guess it would be overkill to try put the `o_dirty` marker specifically on the header or the footer - and it would not really be correct given that the changes are only "about them" but not actually "on them". Not sure if you want to add a "Steps to reproduce" in the commit message - given it is already very well described.

qsm-odoo reviewed (changes requested):
@bvr-odoo I rebased and push-forced to add [this change](https://github.com/odoo/odoo/compare/74ce59a94cec5ed4d65ea0571984679632826d3a..cbcb4b490c188ee6be9fbef2673c8fa929fa6778) (so don't forget to fetch to keep these changes) (-> checking the #wrap was ok but I prefer to be a bit more explicit and generic to handle potential custo not having the #wrap element on all pages, also added the "if" to be safer in stable). I guess we can find a better way to mark dirty elements which are not view parts in master.

However, something very important is not ok with this PR: your test does not fail without your fix. So it means that if we rebreak the feature, the test won't help us see it :grimacing: (the test is still nice as it is testing steps that were not tested before but it does not test what is being fixed here). Could you investigate? :+1:

Also to fix: the commit you mention in your commit message is not the merged one:
![image](https://user-images.githubusercontent.com/10338094/178484933-06959f4c-046f-4c06-911b-ebe0ed1ad3ab.png)

bvr-odoo commented:
> @bvr-odoo I rebased and push-forced to add [this change](https://github.com/odoo/odoo/compare/74ce59a94cec5ed4d65ea0571984679632826d3a..cbcb4b490c188ee6be9fbef2673c8fa929fa6778) (so don't forget to fetch to keep these changes) (-> checking the #wrap was ok but I prefer to be a bit more explicit and generic to handle potential custo not having the #wrap element on all pages, also added the "if" to be safer in stable). I guess we can find a better way to mark dirty elements which are not view parts in master.
> 
> However, something very important is not ok with this PR: your test does not fail without your fix. So it means that if we rebreak the feature, the test won't help us see it grimacing (the test is still nice as it is testing steps that were not tested before but it does not test what is being fixed here). Could you investigate? +1
> 
> Also to fix: the commit you mention in your commit message is not the merged one: ![image](https://user-images.githubusercontent.com/10338094/178484933-06959f4c-046f-4c06-911b-ebe0ed1ad3ab.png)

@qsm-odoo Ok done, actually I added the `topMenuColor` at the end but this one makes a change in the wrapwrap.

qsm-odoo commented:
Strange runbot error

@robodoo retry
